packages/cardano-metamask-snap/src/api/cardano__signTransaction/ui/collateral.ts

```diff
@@ -0,0 +1,52 @@
+import { heading, row, text } from '@metamask/snaps-sdk';
+
+import { getUiAccountIndex, section } from '../../ui';
+import { ADA_TICKER, assetValue } from './utils';
+import { OwnAddress, getAddressAccountIndex } from '../../address';
+
+export type CollateralReturnOutput = { address: string; coin: string } & (
+  | { isOwn: false }
+  | {
+      isOwn: true;
+      ownAddress: OwnAddress;
+    }
+);
+
+export type Collateral = {
+  collateralReturn: CollateralReturnOutput | undefined;
+  totalCollateral: string | undefined;
+};
+
+export const renderCollateralReturn = (
+  collateralReturnOutput: CollateralReturnOutput,
+) => {
+  const amountRow = row(
+    'Amount',
+    text(assetValue(collateralReturnOutput.coin, ADA_TICKER)),
+  );
+  const accountIndex = collateralReturnOutput.isOwn
+    ? getAddressAccountIndex(collateralReturnOutput.ownAddress)
+    : undefined;
+
+  if (accountIndex) {
```

Comment:
as long as collateral is to be returned to user's own address, I wouldn't bother the user with that info as they just get the funds back. I think we already deal with change outputs in a similar way